crowbar_framework/app/models/dns_service.rb

```diff
@@ -67,20 +67,21 @@ def transition(inst, name, state)
     #
     if state == "discovered"
       @logger.debug("DNS transition: handling for #{name} for #{state}: discovered")
-      db = Proposal.where(barclamp: "dns", name: inst).first
       role = RoleObject.find_role_by_name "dns-config-#{inst}"
 
-      if role.default_attributes["dns"]["auto_assign_server"]
-        if role.override_attributes["dns"]["elements"]["dns-server"].nil? or
-           role.override_attributes["dns"]["elements"]["dns-server"].empty?
-          @logger.debug("DNS transition: adding #{name} to dns-server role")
-          result = add_role_to_instance_and_node("dns", inst, name, db, role, "dns-server")
+      Proposal.where(barclamp: "dns", name: inst).first do |db|
+        if role.default_attributes["dns"]["auto_assign_server"]
+          if role.override_attributes["dns"]["elements"]["dns-server"].nil? or
+            role.override_attributes["dns"]["elements"]["dns-server"].empty?
+            @logger.debug("DNS transition: adding #{name} to dns-server role")
+            result = add_role_to_instance_and_node("dns", inst, name, db, role, "dns-server")
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `result`. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)